go.mod

```diff
@@ -1,29 +1,29 @@
 module github.com/openshift/cluster-authentication-operator
 
-go 1.20
+go 1.21
 
 require (
 	github.com/davecgh/go-spew v1.1.1
 	github.com/ghodss/yaml v1.0.0
 	github.com/go-bindata/go-bindata v3.1.2+incompatible
-	github.com/google/go-cmp v0.5.9
-	github.com/openshift/api v0.0.0-20231010075512-1ccc6058c62d
-	github.com/openshift/build-machinery-go v0.0.0-20220913142420-e25cf57ea46d
-	github.com/openshift/client-go v0.0.0-20230926161409-848405da69e1
-	github.com/openshift/library-go v0.0.0-20231128230659-785a9313da6c
+	github.com/google/go-cmp v0.6.0
+	github.com/openshift/api v0.0.0-20231219140051-ddc590a81acb
```

Comment:
pinned to slightly older version as there seems to be an issue with pulling https://github.com/openshift/api/tree/master/operator/v1/zz_generated.crd-manifests used in https://github.com/openshift/cluster-authentication-operator/blob/master/Dockerfile.rhel7#L10

I think that `go mod vendor` only incudes files required to compile the binary, since we don't reference these manifest they are not added to the vendor directory.

note that having the newest version of the api repo is not directly required by this PR.